### ISSUE TITLE
"onClientPlayerDamage" Should return the amount of pellets that hit

### ISSUE THREAD

**Issue by dinamoMTA** [multitheftauto/mtasa-blue#644]
**Is your feature request related to a problem? Please describe.**
I'm making a custom damage script, but I cannot difference when a player in a nearby distance shoots other player with a shotgun if he has hit with all the pellets or just with some of them.
By returning the amount of pellets that hit I wouldn't have any problem because I could just multiply the damage I want to do with the pellets so the system would work perfectly but I just can't...

**Describe the solution you'd like**
"onClientPlayerDamage" would return the amount of pellets that hit the enemy player

**Describe alternatives you've considered**
I have tried calculating with screen position the amount of damage it should do but it's just inefficient, not only I have to use kind of complex trigonometry operations also the result isn't as good as if it would just give me that amount of pellets that hit.

**Additional context**
If you did not understand well what I said I mean that the event should difference this:
![](https://i.imgur.com/DoiPtYa.jpg)
From this
![](https://i.imgur.com/zJVJECp.jpg)

**Comment by PwN3RB0T** [comment]
A good workaround is to lookup the damage-per bullet and then divide it with the amount of damage givent..this is at least how it's done in SA-MP

**Comment by boycsuk** [comment]
I think that this feature would be helpful, because I'm very interested on enhancing my server PvP capabilities, MTA's actual sync is not very reliable with guns and this feature would be very useful in vast of the majority PvP oriented servers such like TDM ones or even RPG servers, not only exceptionally for me

**Comment by dretax** [comment]
Yeah. It was also hard for me to get shotguns work for my realistic weapon and armor systems.
